doc/migrations/v0.27-v.28.md

```diff
@@ -0,0 +1,343 @@
+
+`libp2p.create` receives a `peerId` property instead of a `peerInfo` property.
+
+**Before**
+```js
+const peerId = ...
+const peerInfo = new PeerInfo(peerId)
+
+const libp2p = await Libp2p.create({
+  peerInfo
+  // ...
+})
+```
+
+**After**
+```js
+const peerId = ...
+
+const libp2p = await Libp2p.create({
+  peerId
+  // ...
+})
+```
+
+#### Provide listen addresses for transport listen
+
+**Before**
+```js
+const peerId = ...
+const peerInfo = new PeerInfo(peerId)
+
+peerInfo.multiaddrs.add('/ip4/127.0.0.1/tcp/0')
+
+const libp2p = await Libp2p.create({
+  peerInfo
+  // ...
+})
+
+await libp2p.start()
+```
+
+or
+
+```js
+const peerId = ...
+const peerInfo = new PeerInfo(peerId)
+
+const libp2p =  await Libp2p.create({
+  peerInfo
+  // ...
+})
+
+libp2p.peerInfo.multiaddrs.add('/ip4/127.0.0.1/tcp/0')
+
+await libp2p.start()
+```
+
+**After**
+```js
+const peerId = ...
+
+const libp2p = await Libp2p.create({
+  peerId,
+  addresses: {
+    listen: ['/ip4/127.0.0.1/tcp/0']
+  }
+  // ...
+})
+await libp2p.start()
+```
+
+There is also a bonus regarding the peer addresses. `libp2p@0.28` comes with an AddressManager that also allows the configuration of `announce` and `noAnnounce` addresses.
+This was possible to achieve before, but in a hacky way by removing or adding addresses to the `peerInfo`, after the node starts.
+
+**Before**
+```js
+const peerId = ...
+const peerInfo = new PeerInfo(peerId)
+
+peerInfo.multiaddrs.add('/ip4/127.0.0.1/tcp/8000')
+
+const libp2p = await Libp2p.create({
+  peerInfo
+  // ...
+})
+
+await libp2p.start()
+peerInfo.multiaddrs.add('/dns4/peer.io') // Announce
+peerInfo.multiaddrs.delete('/ip4/127.0.0.1/tcp/8000') // Not announce
+```
+
+**After**
+```js
+const peerId = ...
+
+const libp2p = await Libp2p.create({
+  peerId,
+  addresses: {
+    listen: ['/ip4/127.0.0.1/tcp/0'],
+    announce: ['/dns4/peer.io'],
+    noAnnounce: ['/ip4/127.0.0.1/tcp/0']
+  }
+  // ...
+})
+await libp2p.start()
+```
+
+### API Implications
+
+#### Peer Dialing, Hangup and Ping
+
+`libp2p.dial`, `libp2p.dialProtocol`, `libp2p.hangup` and `libp2p.ping` supported as the target parameter a [`PeerInfo`](peer-info), a [`PeerId`](peer-id), a [`Multiaddr`][multiaddr] and a string representation of the multiaddr. Considering that [`PeerInfo`](peer-info) is being removed from libp2p, all these methods will now support the other 3 possibilities. 
+
+There is one relevant aspect to consider with this change. When using a [`PeerId`](peer-id), the PeerStore will need to have known addresses for that peer in its AddressBook, so that it can fulfil the request. This was also true in the past, but it is important pointing it out because it might not be enough to switch from using [`PeerInfo`](peer-info) to [`PeerId`](peer-id). When using a [`PeerInfo`](peer-info), the PeerStore was not required to have the multiaddrs for the peer since libp2p was inspecting its multiaddrs, in order to use them to reach other peers.
```

Comment:
```suggestion
There is one relevant aspect to consider with this change. When using a [`PeerId`](peer-id), the PeerStore **MUST** have known addresses for that peer in its AddressBook, so that it can perform the request. This was also true in the past, but it is important pointing it out because it might not be enough to switch from using [`PeerInfo`](peer-info) to [`PeerId`](peer-id). When using a [`PeerInfo`](peer-info), the PeerStore was not required to have the multiaddrs when they existed on the PeerInfo instance.
```